src/Simplify.elm

```diff
@@ -846,7 +847,10 @@ type Configuration
 -}
 defaults : Configuration
 defaults =
-    Configuration { ignoreConstructors = [] }
+    Configuration
+        { ignoreConstructors = []
+        , expectNaN = False
```

Comment:
This is I guess what you wanted feedback on?
It's a really hard decision because we want to balance trustable, bug-free refactorings with convenience.
I like that users can decide to tell simplify that they don't expect NaN and get the convenience. NaN is used almost never so the trade-off to require explicit confirmation that NaNs are used makes sense.

What we do need in my opinion is a very clear indication in the documentation that you should decide whether you want to set this option for your whole project.
(If I was writing the rule, I'd probably make the convenience opt-in, so that users always explicitly know of the trade-offs and consent to them but good documentation should do the job just as well.)